In [1]:
# 1. Installing the required tools,   softwares, libraries & packages
!apt-get install -y sra-toolkit
!apt-get install -y fastqc
!apt-get install -y trimmomatic
!pip install cutadapt


Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following additional packages will be installed:
  blends-common libkdf5-2 libncbi-vdb2 libncbi-wvdb2 med-config menu
Suggested packages:
  blends-doc menu-l10n gksu | kde-runtime | ktsuss
The following NEW packages will be installed:
  blends-common libkdf5-2 libncbi-vdb2 libncbi-wvdb2 med-config menu
  sra-toolkit
0 upgraded, 7 newly installed, 0 to remove and 38 not upgraded.
Need to get 8,290 kB of archives.
After this operation, 23.0 MB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 menu amd64 2.1.47ubuntu4 [354 kB]
Get:2 http://archive.ubuntu.com/ubuntu jammy/universe amd64 blends-common all 0.7.4ubuntu1 [15.9 kB]
Get:3 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libkdf5-2 amd64 2.11.2+dfsg-4build2 [14.7 kB]
Get:4 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libncbi-vdb2 amd64 2.11.2+dfsg-4build2 [1,364 kB]
Get

In [2]:
# 2. Verifying the tool version's
!fastqc -v
!trimmomatic -version
!cutadapt --version
!fastq-dump --version


FastQC v0.11.9
/bin/bash: line 1: trimmomatic: command not found
5.2

"fastq-dump" version 2.11.3



In [3]:
# 3. Removing the incorrectly installed Trimmomatic
!rm -rf /usr/local/bin/trimmomatic*
!rm -rf trimmomatic-0.39.zip Trimmomatic-0.39

In [4]:
# 4. Installing the Trimmomatic Tool
!wget -q https://github.com/usadellab/Trimmomatic/releases/download/v0.39/trimmomatic-0.39.zip
!unzip -q trimmomatic-0.39.zip

In [5]:
# 5. Move to Safer location
!mv Trimmomatic-0.39 /usr/local/bin/

# 6. Creating an executable wrapper script
!echo 'java -jar /usr/local/bin/Trimmomatic-0.39/trimmomatic-0.39.jar "$@"' > /usr/local/bin/trimmomatic
!chmod +x /usr/local/bin/trimmomatic

#  Verfiying Trmimmomatic Version
!trimmomatic -version

0.39


In [6]:
# 7. Checking the fastq Version.
!fastq-dump --version


"fastq-dump" version 2.11.3



In [7]:
# 8. Making a directory
!mkdir -p /content/sra_data
%cd /content/sra_data

/content/sra_data


In [8]:
# 9. Downloading  directly from NCBI & Splitting  paired-end reads
!fastq-dump --split-files SRR390728 --gzip

Read 7178576 spots for SRR390728
Written 7178576 spots for SRR390728


In [9]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [10]:
#10 Listing the Contents in the SRA data
!ls -1h /content/sra_data

SRR390728_1.fastq.gz
SRR390728_2.fastq.gz


In [11]:
# 11. Viewing the header files in the SRA data
!zcat SRR390728_1.fastq.gz | head -n 8

@SRR390728.1 1 length=36
CATTCTTCACGTAGTTCTCGAGCCTTGGTTTTCAGC
+SRR390728.1 1 length=36
;;;;;;;;;;;;;;;;;;;;;;;;;;;9;;665142
@SRR390728.2 2 length=36
AAGTAGGTCTCGTCTGTGTTTTCTACGAGCTTGTGT
+SRR390728.2 2 length=36
;;;;;;;;;;;;;;;;;4;;;;3;393.1+4&&5&&


In [12]:
# 12. Making a directory for the fastqc
!mkdir -p /content/fastqc_pre

In [13]:
# 13. Checking the Quality Control of Splitted files
!fastqc SRR390728_1.fastq.gz SRR390728_2.fastq.gz -o /content/fastqc_pre

Started analysis of SRR390728_1.fastq.gz
Approx 5% complete for SRR390728_1.fastq.gz
Approx 10% complete for SRR390728_1.fastq.gz
Approx 15% complete for SRR390728_1.fastq.gz
Approx 20% complete for SRR390728_1.fastq.gz
Approx 25% complete for SRR390728_1.fastq.gz
Approx 30% complete for SRR390728_1.fastq.gz
Approx 35% complete for SRR390728_1.fastq.gz
Approx 40% complete for SRR390728_1.fastq.gz
Approx 45% complete for SRR390728_1.fastq.gz
Approx 50% complete for SRR390728_1.fastq.gz
Approx 55% complete for SRR390728_1.fastq.gz
Approx 60% complete for SRR390728_1.fastq.gz
Approx 65% complete for SRR390728_1.fastq.gz
Approx 70% complete for SRR390728_1.fastq.gz
Approx 75% complete for SRR390728_1.fastq.gz
Approx 80% complete for SRR390728_1.fastq.gz
Approx 85% complete for SRR390728_1.fastq.gz
Approx 90% complete for SRR390728_1.fastq.gz
Approx 95% complete for SRR390728_1.fastq.gz
Analysis complete for SRR390728_1.fastq.gz
Started analysis of SRR390728_2.fastq.gz
Approx 5% complete fo

In [14]:
# 14. Listing the contents
!ls -1h /content/fastqc_pre


SRR390728_1_fastqc.html
SRR390728_1_fastqc.zip
SRR390728_2_fastqc.html
SRR390728_2_fastqc.zip


In [15]:
# 15. Downnloading the files
from google.colab import files
files.download("/content/fastqc_pre/SRR390728_1_fastqc.html")
files.download("/content/fastqc_pre/SRR390728_2_fastqc.html")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [16]:
# 16. Preparing the  Adapter File for Trimmomatic
!mkdir -p /usr/local/bin/Trimmomatic-0.39/adapters
!wget -q https://raw.githubusercontent.com/timflutre/trimmomatic/master/adapters/TruSeq3-PE.fa -O /usr/local/bin/Trimmomatic-0.39/adapters/TruSeq3-PE.fa


In [17]:
# 17. Trimming  Reads with Trimmomatic tool.
!mkdir -p /content/trimmed
!trimmomatic PE -threads 2 \
SRR390728_1.fastq.gz SRR390728_2.fastq.gz \
/content/trimmed/SRR390728_1_paired.fq.gz /content/trimmed/SRR390728_1_unpaired.fq.gz \
/content/trimmed/SRR390728_2_paired.fq.gz /content/trimmed/SRR390728_2_unpaired.fq.gz \
ILLUMINACLIP:/usr/local/bin/Trimmomatic-0.39/adapters/TruSeq3-PE.fa:2:30:10 \
LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36


TrimmomaticPE: Started with arguments:
 -threads 2 SRR390728_1.fastq.gz SRR390728_2.fastq.gz /content/trimmed/SRR390728_1_paired.fq.gz /content/trimmed/SRR390728_1_unpaired.fq.gz /content/trimmed/SRR390728_2_paired.fq.gz /content/trimmed/SRR390728_2_unpaired.fq.gz ILLUMINACLIP:/usr/local/bin/Trimmomatic-0.39/adapters/TruSeq3-PE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36
Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'
ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences
Quality encoding detected as phred33
Input Read Pairs: 7178576 Both Surviving: 3612785 (50.33%) Forward Only Surviving: 810889 (11.30%) Reverse Only Surviving: 1886806 (26.28%) Dropped: 868096 (12.09%)
TrimmomaticPE: Completed successfully


In [18]:
# 18. Trimming  Reads with Cutadapt tool
!mkdir -p /content/cutadapt_out
!cutadapt -a AGATCGGAAGAGC -A AGATCGGAAGAGC \
-o /content/cutadapt_out/SRR390728_1_trimmed.fq.gz \
-p /content/cutadapt_out/SRR390728_2_trimmed.fq.gz \
SRR390728_1.fastq.gz SRR390728_2.fastq.gz


This is cutadapt 5.2 with Python 3.12.12
Command line parameters: -a AGATCGGAAGAGC -A AGATCGGAAGAGC -o /content/cutadapt_out/SRR390728_1_trimmed.fq.gz -p /content/cutadapt_out/SRR390728_2_trimmed.fq.gz SRR390728_1.fastq.gz SRR390728_2.fastq.gz
Processing paired-end reads on 1 core ...
Done           00:01:22     7,178,576 reads @  11.5 µs/read;   5.23 M reads/minute

=== Summary ===

Total read pairs processed:          7,178,576
  Read 1 with adapter:                 234,022 (3.3%)
  Read 2 with adapter:                 242,591 (3.4%)
Pairs written (passing filters):     7,178,576 (100.0%)

Total basepairs processed:   516,857,472 bp
  Read 1:   258,428,736 bp
  Read 2:   258,428,736 bp
Total written (filtered):    514,199,470 bp (99.5%)
  Read 1:   257,122,256 bp
  Read 2:   257,077,214 bp

=== First read: Adapter 1 ===

Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 234022 times

Minimum overlap: 3
No. of allowed errors:
1-9 bp: 0; 10-13 bp: 1

Bases preceding remov

# Here's an interpretation of the findings from the outputs you generated:

Initial Quality Control (FastQC):

The FastQC reports generated in step 13 give you an initial assessment of the raw sequencing data quality. You would typically open the downloaded HTML files (SRR390728_1_fastqc.html and SRR390728_2_fastqc.html) to see detailed plots and metrics. Key things to look for in these reports include:

    Per Base Sequence Quality: This plot shows the quality scores for each base position in the reads. You want to see high quality scores across the length of the reads. Low quality at the ends is common and addressed by trimming.
    Per Sequence Quality Scores: This plot shows the distribution of average quality scores per read. A good library should have a peak towards the higher quality scores.
    Per Base Sequence Content: This plot shows the proportion of each base (A, T, G, C) at each position. A skewed distribution at the beginning of the reads can indicate a problem with library preparation.
    Adapter Content: This module checks for the presence of common sequencing adapters. If adapters are present, they need to be trimmed, which you addressed in the subsequent steps.
    Sequence Length Distribution: This shows the distribution of read lengths.

Trimming with Trimmomatic and Cutadapt:

You used two different tools, Trimmomatic and Cutadapt, for trimming adapters and low-quality bases.

    Trimmomatic (Step 17): The output shows that Trimmomatic processed 7,178,576 read pairs.
        50.33% of the read pairs had both reads surviving the trimming process.
        11.30% had only the forward read surviving.
        26.28% had only the reverse read surviving.
        12.09% of the read pairs were dropped entirely because both reads were too short after trimming or failed other criteria. This indicates that a significant portion of the reads required trimming, and some reads or read pairs were removed due to poor quality or being too short.
    Cutadapt (Step 18): The Cutadapt output confirms that it processed the same number of read pairs (7,178,576).
        It identified and trimmed adapters from 3.3% of Read 1 and 3.4% of Read 2.
        Importantly, it reported that 100% of the read pairs were written after filtering (passing filters). This means that while adapters were trimmed, Cutadapt did not discard entire read pairs based on the adapter trimming alone, unlike Trimmomatic's various filtering steps (LEADING, TRAILING, SLIDINGWINDOW, MINLEN).
        The total base pairs written after trimming is slightly less than the total processed, indicating that bases were removed due to adapter trimming.

Summary of Findings:

The initial FastQC reports (which you would view separately) likely showed the presence of adapters and potentially some low-quality bases, especially towards the ends of the reads. Both Trimmomatic and Cutadapt successfully identified and removed these adapters. Trimmomatic performed additional quality filtering based on the criteria you specified (leading/trailing low quality, sliding window, minimum length), resulting in a percentage of reads being dropped. Cutadapt focused specifically on adapter trimming and reported trimming a small percentage of reads, writing all pairs after trimming.

The trimmed reads in the /content/trimmed and /content/cutadapt_out directories are now cleaner and more suitable for downstream analysis, such as alignment to a reference genome or transcriptome. The choice of which trimmed output to use (from Trimmomatic or Cutadapt) depends on your specific analysis goals and how aggressively you want to filter based on quality.